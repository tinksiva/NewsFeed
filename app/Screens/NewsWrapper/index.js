import { SafeAreaView } from "react-native";
import { useCallback, useRef, useState } from "react";
import NewsFeed from "../NewsFeed/index";
import { fetchFromApi, storeInAsync } from "../../utils/utils";
import { globalStyles } from "../../utils/globalStyles";

export default function NewsWrapper() {
  //loading is an indicator to display the indicator while we fetch the data
  const page = useRef(1);
  const refetch = useCallback(async () => {
    let data = await fetchFromApi(page.current);
    let response = false;
    if (!data) {
      response = false;
    } else {
      response = await storeInAsync(data);
    }
    page.current = page.current + 1;
    return response;
  }, []);
  return (
    <SafeAreaView style={globalStyles.flex1}>
      <NewsFeed refetch={refetch} />
    </SafeAreaView>
  );
}
