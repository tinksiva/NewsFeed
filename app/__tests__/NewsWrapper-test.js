jest.useFakeTimers();
import "react-native";
import React from "react";
import NewsWrapper from "../Screens/NewsWrapper";
import { create, act } from "react-test-renderer";
jest.mock("@react-native-community/netinfo", () => ({
  fetch: () => Promise.resolve(jest.fn()),
  addEventListener: () => {
    return () => {};
  },
}));
jest.mock("./../utils/utils", () => {
  return {
    fetchFromApi: jest.fn().mockImplementation((page = 1) => {
      return Promise.resolve(Object);
    }),
    storeInAsync: jest.fn().mockImplementation((Object) => {
      return Promise.resolve(true);
    }),
  };
});

it("renders correctly", () => {
  let tree;
  act(() => {
    tree = create(<NewsWrapper />);
  });
  tree = tree.toJSON();
  expect(tree).toMatchSnapshot();
});
