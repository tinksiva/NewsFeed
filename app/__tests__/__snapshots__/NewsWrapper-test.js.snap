// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#D3D3D333",
        "flex": 1,
        "paddingBottom": "1%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "black",
          "height": "8%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Times New Roman",
            "fontSize": 9,
            "fontWeight": "bold",
            "marginTop": 60,
            "textAlign": "center",
          }
        }
      >
        Headlines
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": "2%",
        }
      }
    >
      <RCTScrollView
        ListEmptyComponent={
          <Memo(EmptyList)
            error={false}
          />
        }
        contentContainerStyle={
          {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        data={[]}
        extraData={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        getItemLayout={[Function]}
        initialNumToRender={7}
        keyExtractor={[Function]}
        maxToRenderPerBatch={5}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        style={
          {
            "flex": 1,
            "marginBottom": "12.5%",
          }
        }
        viewabilityConfigCallbackPairs={[]}
        windowSize={15}
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "width": "60%",
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "center",
                  "textAlign": "center",
                }
              }
            >
              Please Wait while we fetch the latest news for you
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
